rakelib/output.rake

```diff
@@ -0,0 +1,94 @@
+if File.directory? 'Tests/Expected'
+  MODULE_INPUT_PATH = 'Fixtures/stub-env/Modules'
+  MODULE_OUTPUT_PATH = 'Fixtures/stub-env'
+  SDKS = {
+    :macosx => 'x86_64-apple-macosx10.12',
+    :iphoneos => 'armv7s-apple-ios10.0'
+  }
+  TOOLCHAINS = {
+    :swift2 => {
+      :module_path => "#{MODULE_OUTPUT_PATH}/swift2.3",
+      :toolchain => 'com.apple.dt.toolchain.Swift_2_3'
+    },
+    :swift3 => {
+      :module_path => "#{MODULE_OUTPUT_PATH}/swift3",
+      :toolchain => 'com.apple.dt.toolchain.XcodeDefault'
+    }
+  }
+
+  namespace :output do
+    desc 'Compile modules'
+    task :modules do |task|
+      Utils.print_info 'Compile output modules'
+
+      # macOS
+      modules = ['PrefsWindowController']
+      modules.each do |m|
+        puts "Compiling module #{m}… (macos)"
+        compile_module(m, :macosx, task)
+      end
+
+      # iOS
+      modules = ['CustomSegue', 'LocationPicker', 'SlackTextViewController']
+      modules.each do |m|
+        puts "Compiling module #{m}… (ios)"
+        compile_module(m, :iphoneos, task)
+      end
+
+      # delete swiftdoc
+      Dir.glob("#{MODULE_OUTPUT_PATH}/*.swiftdoc").each do |f|
+        FileUtils.rm_rf(f)
+      end
+    end
+
+    desc 'Compile output'
+    task :compile => :modules do |task|
+      Utils.print_info 'Compiling template output files'
+
+      exit Dir.glob('Tests/Expected/**/*.swift').map { |f|
```

Comment:
`exit` typically takes an integer (the exit code) as argument, right?

So if one of the `compile_file(f, task)` returns `false`, the `reduce` will return `false` and `exit false` would be interpreted as `exit(0)`… generally meaning no error?
While if all `compile_file(f, task)` return `true`, the`reduce` will return `true` and `exit true` would be interpreted as `exit(1)`… meaning an error?

I think it would be better if `compile_file` returned the return code / status code of the execution of the `Utils.run` call (line 87), and doing a `exit_codes = Dir.glob(…).map { compile_file(…) }` first, then do a `return exit_codes.find { |c| c != 0 } || 0` to globally return the first non-zero exit code, or exit(0) if all exit codes are 0.